compiler-cli/src/main.rs

```diff
@@ -520,7 +520,10 @@ fn initialise_logger() {
 }
 
 fn project_paths_at_current_directory() -> ProjectPaths {
-    let current_dir = std::env::current_dir().expect("Could not get current directory");
+    let current_dir = Utf8PathBuf::from_path_buf(
+        std::env::current_dir().expect("Could not get current directory"),
+    )
+    .expect("Non Utf-8 Path");
```

Comment:
This code exists in a few places, let's move it to a helper function in `fs`.

Reply:
We use this in compiler-core as well. Are you okay if I put the helper in compiler-core::io?

Reply:
Oh yikes, it shouldn't be in there- it performs IO. We'll have to move it to the IO trait, or at least put a huge TODO: FIXME on it for now

Reply:
I think this might involve a more serious think haha. I think a big TODO fix me (with an accompanying issue) would be a good idea. Compiler core has a concept of DirEntry, ReadDir, OutputFile, FileIOError, ProjectPaths, etc. all based on paths with real implementations instead of just traits. There could be a clever way to extract all this stuff to be based on a simple enough trait for compiler-cli and others to implement if you really want compiler-core's conception of a file and a path to be abstract.

Reply:
> all based on paths with real implementations instead of just traits

What do you mean by this?

Reply:
OH I meant converting from path to utf8 path in general (which is in the language server and paths. Language server needs to convert from lsp crates `Url` type, paths needs to convert from `dirs_next::cache_dir` call).

Reply:
Oh right! Anything pure is fine into do in core 👍 

Sounds good

Reply:
Done :)